src/runtime/program/vote/state.zig

```diff
@@ -132,7 +476,100 @@ pub const VoteState = struct {
         return 3762;
     }
 
+    /// https://github.com/anza-xyz/solana-sdk/blob/4e30766b8d327f0191df6490e48d9ef521956495/vote-interface/src/state/mod.rs#L862
+    pub fn setNewAuthorizedVoter(
+        self: *VoteState,
+        new_authorized_voter: Pubkey,
+        target_epoch: Epoch,
+    ) (InstructionError || VoteError)!void {
+
+        // The offset in slots `n` on which the target_epoch
+        // (default value `DEFAULT_LEADER_SCHEDULE_SLOT_OFFSET`) is
+        // calculated is the number of slots available from the
+        // first slot `S` of an epoch in which to set a new voter for
+        // the epoch at `S` + `n`
+        if (self.authorized_voters.contains(target_epoch)) {
+            return VoteError.TooSoonToReauthorize;
+        }
+
+        const epoch, const pubkey = self.authorized_voters.last() orelse
+            return InstructionError.InvalidAccountData;
+
+        if (!pubkey.equals(&new_authorized_voter)) {
+            const epoch_of_last_authorized_switch = if (self.prior_voters.last()) |prior_voter|
+                prior_voter.end
+            else
+                0;
+
+            if (target_epoch <= epoch) {
+                return InstructionError.InvalidAccountData;
+            }
+
+            self.prior_voters.append(PriorVote{
+                .key = pubkey,
+                .start = epoch_of_last_authorized_switch,
+                .end = target_epoch,
+            });
+        }
+
+        self.authorized_voters.insert(target_epoch, new_authorized_voter) catch
+        // TODO: Is it okay to convert out of memory to InvalidAccountData?
```

Comment:
doesnt make sense to me, do we need only instruction errors as return types?

Reply:
[5b20988](https://github.com/Syndica/sig/pull/576/commits/5b20988e3e7b71b6b3267bf464732fdfcf9c203b)